mountpoint-s3/src/inode.rs

```diff
@@ -1042,12 +1105,16 @@ pub enum InodeError {
     ShadowedByDirectory(String, InodeNo),
     #[error("inode {0} is not a directory")]
     NotADirectory(InodeNo),
+    #[error("inode {0} is a directory")]
```

Comment:
What is meant by inode {0} here?

Reply:
It is performing string formatting / interpolation.

The `0` is referring to the index of the value in the `IsDirectory` enum variant - in this case, the `InodeNo` on the line below.

This attribute is used by macros in the crate `thiserror`. You can see a quick summary of what this turns into under the "A Display impl is generated for your error" bullet point. https://docs.rs/thiserror/latest/thiserror/#details